accounting/transferwise/tests/test_models.py

```diff
@@ -0,0 +1,153 @@
+        """ `create_payments` is called when `auto_create_payments_on_save` is checked. """
+        self.bulk_payment.auto_create_payments_on_save = True
+        self.bulk_payment.save()
+        mock_create_payments.assert_called_once()
+
+    @mock.patch.object(models.TransferWiseBulkPayment, 'to_bulk_payment_csv')
+    def test_auto_create_csv_on_save(self, mock_to_bulk_payment_csv):
+        """ `to_bulk_payment_csv` is called when `auto_create_csv_on_save` is checked. """
+        self.bulk_payment.auto_create_csv_on_save = True
+        self.bulk_payment.save()
+        mock_to_bulk_payment_csv.assert_called_once()
+
+    @mock.patch.object(models.TransferWiseBulkPayment, 'upload_to_google_drive')
+    def test_auto_upload_google_drive_on_save(self, mock_upload_to_google_drive):
+        """ `upload_to_google_drive` is called when `auto_upload_google_drive_on_save` is checked. """
+        csv_path = 'https://drive.google.com/bulk_payment.csv'
+        mock_upload_to_google_drive.return_value = {'webContentLink': csv_path}
+        self.bulk_payment.auto_upload_google_drive_on_save = True
+        self.bulk_payment.save()
+        mock_upload_to_google_drive.assert_called_once()
+        self.assertEqual(self.bulk_payment.csv_path, csv_path)
+
+    def _create_recipients(self, invoices=False, invoice_date=NOW):
+        """
+        Create recipients for this bulk transfer, and optionally their invoices.
+
+        :param invoices: Whether to create invoices for the recipients.
+        :param invoice_date: What date to use for recipient invoices. Defaults to a frozen date.
+        :return: The list of recipient accounts that were created.
+        """
+        self.provider_bank_account1 = BankAccountFactory(transferwise_recipient_id=1)
+        self.provider_bank_account2 = BankAccountFactory(transferwise_recipient_id=2)
+        HourlyRateFactory(client=self.sender, provider=self.provider_bank_account1.user_account)
+        HourlyRateFactory(client=self.sender, provider=self.provider_bank_account2.user_account)
+        if invoices:
+            InvoiceFactory(date=invoice_date, client=self.sender, provider=self.provider_bank_account1.user_account)
+            InvoiceFactory(date=invoice_date, client=self.sender, provider=self.provider_bank_account2.user_account)
+        return [self.provider_bank_account1.user_account, self.provider_bank_account2.user_account]
+
+
+@ddt.ddt
+class TransferWisePaymentTestCase(TestCase):
+    """ Test cases for `models.TransferWisePayment`. """
+
+    @freezegun.freeze_time(NOW)
+    def setUp(self):
+        """ Setup test objects. """
+        self.provider = AccountFactory(business_name='Developer')
+        self.client = AccountFactory(business_name='OpenCraft GmbH')
+        self.invoice = InvoiceFactory(provider=self.provider, client=self.client)
+        self.bulk_payment = factories.TransferWisePaymentFactory(date=NOW, invoice=self.invoice)
```

Comment:
@UmanShahzad This is for a single payment, so just call this `self.payment`?

Reply:
Oh, that was a bad copy and paste. Jeez I'm bad with this.

Reply:
@UmanShahzad No worries :)